sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryManager.scala

```diff
@@ -353,17 +356,44 @@ class StreamingQueryManager private[sql] (sparkSession: SparkSession) extends Lo
       }
 
       // Make sure no other query with same id is active across all sessions
-      val activeOption =
-        Option(sparkSession.sharedState.activeStreamingQueries.putIfAbsent(query.id, this))
-      if (activeOption.isDefined || activeQueries.values.exists(_.id == query.id)) {
+      val activeOption = Option(sparkSession.sharedState.activeStreamingQueries.get(query.id))
+        .orElse(activeQueries.get(query.id))
+
+      val turnOffOldStream =
+        sparkSession.sessionState.conf.getConf(SQLConf.STOP_RUNNING_DUPLICATE_STREAM)
+      if (activeOption.isDefined && turnOffOldStream) {
+        val oldQuery = activeOption.get
+        logWarning(s"Stopping existing streaming query [id=${query.id}, runId=${oldQuery.runId}]," +
+          " as a new run is being started.")
+        Some(oldQuery)
```

Comment:
It seems that we can stop this query here instead of returning `Option`.
`activeDuplicateQuery` is used only once at `activeDuplicateQuery.foreach(_.stop())`, isn't it?

Reply:
Then, we don't need `val activeDuplicateQuery = ` declararion.